integration-tests/client/test/core.test.ts

```diff
@@ -0,0 +1,355 @@
+    },
+    {
+        name: "val3",
+        mnemonic:
+            "tenant neck ask season exist hill churn rice convince shock modify evidence armor track army street stay light program harvest now settle feed wheat",
+    },
+];
+
+// Initialize accounts and wait for the chain to start
+beforeAll(async () => {
+    console.log("setting up accounts...");
+
+    // @ts-expect-error
+    strideAccounts = {};
+    // @ts-expect-error
+    gaiaAccounts = {};
+
+    for (const { name, mnemonic } of mnemonics) {
+        // setup signer for Stride
+        const signer = await DirectSecp256k1HdWallet.fromMnemonic(mnemonic, {
+            prefix: "stride",
+        });
+
+        const [{ address }] = await signer.getAccounts();
+
+        strideAccounts[name] = await StrideClient.create(
+            STRIDE_RPC_ENDPOINT,
+            signer,
+            address,
+            {
+                gasPrice: GasPrice.fromString(`0.025${USTRD}`),
+                broadcastPollIntervalMs: 50,
+                resolveIbcResponsesCheckIntervalMs: 50,
+            },
+        );
+
+        if (name === "user" || name === "val1") {
+            // setup signer for Gaia
+            const gaiaSigner = await DirectSecp256k1HdWallet.fromMnemonic(mnemonic);
+            const [{ address: gaiaAddress }] = await gaiaSigner.getAccounts();
+
+            gaiaAccounts[name] = {
+                address: gaiaAddress,
+                client: await SigningStargateClient.connectWithSigner(
+                    GAIA_RPC_ENDPOINT,
+                    gaiaSigner,
+                    {
+                        gasPrice: GasPrice.fromString(`1.0${UATOM}`),
+                        broadcastPollIntervalMs: 50,
+                    },
+                ),
+                query: QueryClient.withExtensions(
+                    await Comet38Client.connect(GAIA_RPC_ENDPOINT),
+                    setupAuthExtension,
+                    setupBankExtension,
+                    setupStakingExtension,
+                    setupTxExtension,
+                ),
+            };
+        }
+    }
+
+    console.log("waiting for stride to start...");
+    await waitForChain(strideAccounts.user, USTRD);
+
+    console.log("waiting for gaia to start...");
+    await waitForChain(gaiaAccounts.user, UATOM);
+
+    console.log("waiting for stride-gaia ibc...");
+    await waitForIbc(
+        strideAccounts.user,
+        TRANSFER_CHANNEL.STRIDE.GAIA!,
+        USTRD,
+        "cosmos",
+    );
+}, 45_000);
+
+describe("Core Tests", () => {
+    test("IBC Transfer", async () => {
+        const stridejs = strideAccounts.user;
+        const gaiajs = gaiaAccounts.user;
+        const transferAmount = 50000000;
+
+        // Get initial balances - only track what we can reliably query
+        const { balance: { amount: strideInitialStrideBalance } = { amount: "0" } } =
```

Comment:
Nit: i would rename these so it's clear when you mean the denom vs the chain
```suggestion
        const { balance: { amount: strideInitialStrdBalance } = { amount: "0" } } =
```